snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedMemberOfFieldFixRequest.java

```diff
@@ -0,0 +1,252 @@
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.snomed.datastore.request;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.b2international.snowowl.core.domain.TransactionContext;
+import com.b2international.snowowl.core.events.Request;
+import com.b2international.snowowl.core.exceptions.ComponentNotFoundException;
+import com.b2international.snowowl.core.request.SearchResourceRequestIterator;
+import com.b2international.snowowl.snomed.common.SnomedTerminologyComponentConstants;
+import com.b2international.snowowl.snomed.core.domain.SnomedConcepts;
+import com.b2international.snowowl.snomed.core.domain.SnomedDescriptions;
+import com.b2international.snowowl.snomed.core.domain.refset.SnomedReferenceSet;
+import com.b2international.snowowl.snomed.core.domain.refset.SnomedReferenceSetMembers;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedConceptDocument;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedDescriptionIndexEntry;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedRefSetMemberIndexEntry;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+
+/**
+ * @since 7.21.1
+ */
+public class SnomedMemberOfFieldFixRequest implements Request<TransactionContext, Set<String>> {
+
+	private static final long serialVersionUID = 7930138235619925182L;
+	private static final int LIMIT = 10_000;
+	
+	Set<String> affectedComponentIds = new HashSet<>();
+	
+	@Override
+	public Set<String> execute(TransactionContext context) {
+		Logger log = LoggerFactory.getLogger("dataset-fix-SO-5690");
+		processConcepts(context, log);
+		processDescriptions(context, log);
+		return affectedComponentIds;
+	}
+	
+	private void processConcepts(TransactionContext context, Logger log) {		
+		Set<String> referenceSetIds = SnomedRequests.prepareSearchRefSet()
+				.filterByReferencedComponentType(SnomedTerminologyComponentConstants.CONCEPT)
+				.setLimit(Integer.MAX_VALUE)
+				.build()
+				.execute(context)
+				.stream()
+				.map(SnomedReferenceSet::getId)
+				.collect(Collectors.toSet());
+		
+		log.info("Found a total of " + referenceSetIds.size() + " concept type reference sets");
```

Comment:
Use proper log formatting:

```suggestion
		log.info("Found a total of {} concept type reference sets", referenceSetIds.size());
```